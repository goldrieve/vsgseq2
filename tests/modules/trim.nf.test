nextflow_process {

    name "Test Process TRIM"
    script "modules/trim.nf"
    process "TRIM"

    test("Should trim SE reads") {

        when {
            params {
                cores = 1
            }
            process {
                """
                input[0] = [ [ meta: 'late_se', single_end: true, prefix: 'late_se' ], file("${projectDir}/data/reads/latesingle.fq.gz") ]
                input[1] = params.cores
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
