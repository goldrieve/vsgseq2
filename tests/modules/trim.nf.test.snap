{
    "Should trim SE reads": {
        "content": [
            {
                "0": [
                    [
                        {
                            "meta": "late_se",
                            "single_end": true,
                            "prefix": "late_se"
                        },
                        "late_se_trimmed.fq.gz:md5,ef4a56e3aaa02f34a69a48d82b7000e3"
                    ]
                ],
                "trimmed_reads": [
                    [
                        {
                            "meta": "late_se",
                            "single_end": true,
                            "prefix": "late_se"
                        },
                        "late_se_trimmed.fq.gz:md5,ef4a56e3aaa02f34a69a48d82b7000e3"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-29T13:02:16.246474928"
    }
}